src/main/java/frc/robot/subsystems/Climber.java

```diff
@@ -52,11 +61,32 @@ public Climber() {
     m_rightPIDController.setI(ClimberConstants.kPID.i(), 0);
     m_rightPIDController.setD(ClimberConstants.kPID.d(), 0);
     m_rightPIDController.setFF(0);
+    m_state = State.PRE_FIRST_USE;
+  }
+
+  private boolean canMoveClimber(double position, double speed) {
+    if (
+      m_state == State.AFTER_FIRST_USE &&
+      position <= ClimberConstants.kMinRetractedLength &&
+      Math.signum(speed) == -1.0 ||
+      position >= ClimberConstants.kMaxExtendedLength &&
+      Math.signum(speed) == 1.0
+    ) {
+      return false;
+    }
+    return true;
   }
 
   public void startClimber(double leftSpeed, double rightSpeed) {
-    m_leftPIDController.setReference(leftSpeed, ControlType.kDutyCycle);
-    m_rightPIDController.setReference(rightSpeed, ControlType.kDutyCycle);
+    if (canMoveClimber(m_leftEncoder.getPosition(), leftSpeed)) {
+      m_leftPIDController.setReference(leftSpeed, ControlType.kVelocity);
+    } else {
+      m_leftMotor.stopMotor();
+    }
+    if (canMoveClimber(m_rightEncoder.getPosition(), rightSpeed)) {
+      m_rightPIDController.setReference(rightSpeed, ControlType.kVelocity);
+    }
+    m_rightMotor.stopMotor();
```

Comment:
This sort of logic belongs in `periodic()`. Otherwise nothing assures that we notice when it's unsafe to continue running the motor.